lib/Command/ExpirePassword.php

```diff
@@ -106,54 +129,110 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
-
-		/** @var $user \OCP\IUser */
-		$user = $this->userManager->get($uid);
-
-		if ($user === null) {
-			$output->writeln("<error>Unknown user: $uid</error>");
-			return self::EX_NOUSER;
-		}
-
-		if (!$user->canChangePassword()) {
-			$output->writeln("<error>The user's backend doesn't support password changes. The password cannot be expired for user: $uid</error>");
-			return self::EX_GENERAL_ERROR;
+		$groups = $input->getOption('groups');
+		$allUsers = $input->getOption('all');
+
+		if ($allUsers !== false) {
+			$users = $this->userManager->search('');
+			$this->setPasswordExpiry($input, $output, $users);
+		} elseif (\count($groups) >= 1) {
+			$users = [];
+
+			foreach ($groups as $group) {
+				if ($this->groupManager->groupExists($group) === false) {
+					$output->writeln("Group name $group doesn't exist");
+					return 1;
+				} else {
+					$users[$group] = [];
+					foreach ($this->getAllUsersFromGroup($group) as $users_array) {
+						$users[$group] = $users_array;
+						$this->setPasswordExpiry($input, $output, $users[$group], $group);
+					}
+				}
+			}
+		} else {
+
+			$uid = $input->getArgument('uid');
+
+			/** @var $user \OCP\IUser */
+			$user = $this->userManager->get($uid);
+
+			if ($user === null) {
+				$output->writeln("<error>Unknown user: $uid</error>");
+				return self::EX_NOUSER;
+			}
+
+			if (!$user->canChangePassword()) {
+				$output->writeln("<error>The user's backend doesn't support password changes. The password cannot be expired for user: $uid</error>");
+				return self::EX_GENERAL_ERROR;
+			}
+
+			if ($user !== null) {
+				$this->setPasswordExpiry($input, $output, [$user]);
+			}
 		}
 
-		$expireDate = new \DateTime();
-		$expireDate->setTimezone(new \DateTimeZone('UTC'));
-		$expireDate->setTimestamp($this->timeFactory->getTime());
-		$expireDate->modify($input->getArgument('expiredate'));
-
-		$oldDate = new \DateTime();
-		$oldDate->setTimezone(new \DateTimeZone('UTC'));
-		$oldDate->setTimestamp($this->timeFactory->getTime());
-		$oldDate->modify($input->getArgument('expiredate'));
+		return 0;
+	}
 
-		if ($this->config->getAppValue('password_policy', 'spv_user_password_expiration_checked', false) === 'on') {
-			$delta = $this->config->getAppValue('password_policy', 'spv_user_password_expiration_value', 90);
-			$oldDate->modify("-$delta days");
+	protected function getAllUsersFromGroup($group) {
+		$count = 0;
+		$users = [];
+		foreach ($this->groupManager->findUsersInGroup($group) as $user) {
+			\array_push($users, $user);
+			$count++;
+			//Take 200 users at a time
+			if ($count > 199) {
+				yield $users;
+				$count = 1;
```

Comment:
shouldn't this be 0 ? better write a unit test to confirm that this method works properly when crossing page borders